www/jobs.md

```diff
@@ -4,20 +4,15 @@ The Hail team's mission is to build tools to enable rapid analysis and explorati
 
```

Comment:
@cseed, in the clipped text, it says "10s of TB and tripling yearly" is this still accurate? Aren't we at 100s of TB now?